File: C programming/Competitive/Assignments/Assignment_12/Que_5.c
//question 5: Write a program which accepts a number from user and count how many digits are strictly less than 6.

#include<stdio.h>

int Count(int iNo)
{
    int iDigit = 0, iCnt = 0;

    if(iNo < 0)   // handle negative input
    {
        iNo = -iNo;
    }

    while(iNo != 0)
    {
        iDigit = iNo % 10;   // extract last digit
        if(iDigit < 6)       // check condition
        {
            iCnt++;
        }
        iNo = iNo / 10;      // remove last digit
    }
    return iCnt;
}

int main()
{
    int iValue = 0, iRet = 0;

    printf("Enter number: ");
    scanf("%d",&iValue);

    iRet = Count(iValue);

    printf("Frequency of digits < 6 is: %d\n", iRet);

    return 0;
}

File: Java Programming/Competitve/Assignments/Assignment_20/Que_5.java
//Write a program to find the smallest digit in a given number

class Logic
{
    void findSmallestDigit(int num)
    {
        // Handle 0 as a special case
        if (num == 0)
         {
            System.out.println(0);
            return;
        }

        // Handle negative numbers
        if (num < 0) 
        {
            num = -num;
        }

        int minDigit = 9;

        // Extract digits and find the smallest
        while (num > 0)
         {
            int digit = num % 10;

            if (digit < minDigit) 
            {
                minDigit = digit;
            }
            num = num / 10;
        }

        System.out.println(minDigit);
    }
}

class Que_5
{
    public static void main(String[] args)
     {
         Logic obj = new Logic();
         obj.findSmallestDigit(45872);
        
     }
}
